libgnucash/tax/ca/txf.scm

```diff
@@ -0,0 +1,809 @@
+;; and txf-export-help.scm
+;;
+;; See also https://www.turbotax.com/txf/ [DEAD LINK]
+;;
+;; Updated Jan 2019 to include codes for version 42, although new codes not
+;;   implemented yet because data not available
+;; Updated Feb 2013, Jan 2014 & Jan 2019 . J. Alex Aycinena
+;; Added updated tax line info
+;; Updated Oct 2009. J. Alex Aycinena
+;; Added 'txf-tax-entity-types' and related getter functions
+;; Added 'tax-entity-type' argument to tax code getter functions
+;; Updated txf data for individual tax returns to version 041 (from 039) and
+;;    added tax-line year-effectivity data and code last-year data
+;; Added asset and liability/equity tax code categories
+;; Added version 041 txf data for Partnership, Corporation, S Corporation,
+;;    tax entity types
+;; Added 'None' type for no income tax options
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; This program is free software; you can redistribute it and/or
+;; modify it under the terms of the GNU General Public License as
+;; published by the Free Software Foundation; either version 2 of
+;; the License, or (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program; if not, contact:
+;;
+;; Free Software Foundation           Voice:  +1-617-542-5942
+;; 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652
+;; Boston, MA  02110-1301,  USA       gnu@gnu.org
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define-module (gnucash locale ca tax txf))
+
+(use-modules (gnucash engine))
+(use-modules (gnucash core-utils))
+(use-modules (gnucash app-utils))
+(use-modules (srfi srfi-2))
+(use-modules (gnucash locale ca tax txf-help))
+
+(export gnc:tax-type-txf-get-code-info)
+(export gnc:txf-get-category-key)
+(export gnc:txf-get-code-info)
+(export gnc:txf-get-codes)
+(export gnc:txf-get-description)
+(export gnc:txf-get-form)
+(export gnc:txf-get-format)
+(export gnc:txf-get-help)
+(export gnc:txf-get-last-year)
+(export gnc:txf-get-line-data)
+(export gnc:txf-get-multiple)
+(export gnc:txf-get-payer-name-source)
+(export gnc:txf-get-tax-entity-type)
+(export gnc:txf-get-tax-entity-type-codes)
+(export gnc:txf-get-tax-entity-type-description)
+(export txf-asset-categories)
+(export txf-expense-categories)
+(export txf-help-categories)
+(export txf-income-categories)
+(export txf-liab-eq-categories)
+(export txf-tax-entity-types)
+
+(define txf-tax-entity-types
+  (list
+   (cons 'F1040 #("Individual, Joint, etc." "Files US Form 1040 Tax Return"))
+   (cons 'F1065 #("Partnership" "Files US Form 1065 Tax Return"))
+   (cons 'F1120 #("Corporation" "Files US Form 1120 Tax Return"))
```

Comment:
Surely these don't apply for Canada!

Reply:
lol; removed.

I only know of three types of Canadian business returns (T2125 on a personal return like for sole props, T2, and T2SHORT), which uses these same codes.

I'm having some difficulty building GNUCash (which is why I kept this as a draft). Is there a Docker container that'd be standardized for builds? My Mac doesn't like the dependencies.

Reply:
But what about Canadian personal returns? 

No docker containers that I know about, but there's [this tarball](https://sourceforge.net/projects/gnucash/files/Dependencies/gnucash-5.6-mac-dependencies.tar.xz/download) that we use to set things up for CI. See [macos.yaml](https://github.com/Gnucash/gnucash/blob/stable/.github/workflows/mac-tests.yaml) for how to use it. 

Otherwise you might be able to hack [gnucash-on-osx](https://github.com/gnucash/gnucash-on-osx) to build with Homebrew instead of [gtk-osx](https://gitlab.gnome.org/GNOME/gtk-osx) but you'd have to manually assemble the dependencies. The alternative is [MacPorts](https://wiki.gnucash.org/wiki/MacOS/MacPortsDetail).

Reply:
> But what about Canadian personal returns?

I'm much less concerned about non-sole-prop personal returns (T1 without T2125), as I don't know how often the categorization would come up. I think people just auto-fill their T1s with slips provided to them, and there are not a lot of options for deductions (and even then, unsure how much people would use the field numbers)

Reply:
The general point of TXF is feeding [tax software](https://www.canada.ca/en/revenue-agency/services/e-services/digital-services-individuals/netfile-overview/certified-software-netfile-program.html) instead of hand-keying it. I don't understand why that would be something that only business users would want to do.

Yes, most people with only wage income don't need tax software, but they generally don't need GnuCash either.

Reply:
Other than T2125s, people generally don't hand-key the form fields; they get auto-populated through the slips and [Auto-fill my return](https://www.canada.ca/en/revenue-agency/services/e-services/about-auto-fill-return.html). The additional fields are very few.

Reply:
OK.